core/src/main/java/com/linecorp/armeria/common/stream/AggregationSupport.java

```diff
@@ -160,4 +151,15 @@ private static <U extends AggregatedHttpMessage> U aggregate(List<HttpObject> ob
             return (U) HttpMessageAggregator.aggregateResponse(objects, allocator);
         }
     }
+
+    private static <U extends AggregatedHttpMessage> CompletableFuture<U> handleAggregation(
```

Comment:
One worry about this change is that users won't know whether the returned data is pooled or not.

```
message.aggregate(pooled=true); // returns pooled
message.aggregate(pooled=false); // still returns pooled
```

I'm wondering if we should also store a `boolean` field which indicates whether the cached return value is pooled or not.

Alternatively, we can document more clearly that subscribing with `pooled` is only a hint and doesn't guarantee pooling. In this case, we need to recommend users to always close `HttpData` though which I think is inconvenient 😅

Reply:
> Alternatively, we can document more clearly that subscribing with pooled is only a hint and doesn't guarantee pooling. In this case, we need to recommend users to always close HttpData though which I think is inconvenient 

That's a good suggestion. 👍 
I was actually trying to update the documentation but forgot about that. 😅